sway-lib-std/src/fixed_point.sw

```diff
@@ -0,0 +1,130 @@
+library fixed_point;
+//! A wrapper around U128 type for a library for Sway for mathematical functions operating with signed 64.64-bit fixed point numbers. 
+
+use ::u128::U128;
+use ::assert::assert;
+
+pub struct UFP64 {
+    value: U128 
+}
+
+impl core::ops::Eq for UFP64 {
+    pub fn eq(self, other: Self) -> bool {
+        self.value == other.value
+    }
+}
+
+impl core::ops::Ord for UFP64 {
+    pub fn gt(self, other: Self) -> bool {
+        self.value > other.value
+    }
+
+    pub fn lt(self, other: Self) -> bool {
+        self.value < other.value
+    }
+}
+
+impl core::ops::Add for UFP64 {
+    /// Add a UFP64 to a UFP64. Panics on overflow.
+    pub fn add(self, other: Self) -> Self {
+        UFP64 {
+            value: self.value + other.value
+        }
+    }
+}
+
+impl core::ops::Subtract for UFP64 {
+    /// Subtract a UFP64 from a UFP64. Panics of overflow.
+    pub fn subtract(self, other: Self) -> Self {
+        // If trying to subtract a larger number, panic.
+        assert(!(self.value < other.value));
+
+        UFP64 {
+            value: self.value - other.value
+        }
+    }
+}
+
+impl core::ops::Multiply for UFP64 {
+    /// Multiply a UFP64 with a UFP64. Panics of overflow.
+    pub fn multiply(self, other: Self) -> Self {
+
+        let base = ~U128::from(1,0);
+
+        let self_up = ~U128::from(0, self.value.upper);
+        let self_lo = ~U128::from(0, self.value.lower);
+
+        let other_up = ~U128::from(0, other.value.upper);
+        let other_lo = ~U128::from(0, other.value.lower);
+
+        let mut up_up = self_up * other_up;
```

Comment:
I'd appreciate non-abbreviated names. Ones that are descriptive to any reader who does not have insider knowledge on what is going on